ctfcli/utils/challenge.py

```diff
@@ -404,3 +405,228 @@ def lint_challenge(path):
+    for c in installed_challenges:
+        if c["name"] == challenge["name"]:
+            challenge_id = c["id"]
+            break
+    else:
+        return
+    
+    remote_details = get_challenge_details(challenge_id)
+
+    local_files = {Path(f).name : f for f in challenge.get("files",[])}
+    # Update files
+    for f in remote_details["files"]:
+        # Get base file name
+        f_base = f.split("/")[-1].split('?token=')[0]
+        if f_base not in local_files and create_files:
+            print(f"Creating file {f_base} for challenge {challenge['name']}")
+            # Download remote file and save locally
+            req = s.get(f)
+            req.raise_for_status()
+            Path(challenge.directory, f_base).write_bytes(req.content)
+            if "files" not in challenge:
+                challenge["files"] = []
+            challenge["files"].append(f_base)
+
+        elif f_base in local_files and update_files:
+            # Download remote file and replace local file
+            print(f"Updating file {f_base} for challenge {challenge['name']}")
+            req = s.get(f)
+            req.raise_for_status()
+            remote_file = req.content
+            local_file = Path(challenge.directory, local_files[f_base])
+            local_file.write_bytes(remote_file)
+
+    # Remove files that are no longer present on the remote challenge
+    remote_cleaned_files = [f.split("/")[-1].split('?token=')[0] for f in remote_details["files"]]
+    challenge["files"] = [f for f in challenge.get("files",[]) if Path(f).name in remote_cleaned_files]
+    del remote_details["files"]
+        
+    print(f"Updating challenge.yml for {challenge['name']}")
+
+    # Prefer ordering remote details as described by spec
+    preferred_order = ["name", "category", "description", \
+        "value", "type", "connection_info", "attempts", \
+            "flags", "topics", "tags", "files", "hints", "requirements", "state"]
+
+    # Merge local and remote challenge.yml & Preserve local keys + order
+    updated_challenge = dict(challenge)
+
+    # Ignore optional fields with default values
+    remote_details_updates = {}
+    for k, v in remote_details.items():
+        # If the key value changed, we want to update it
+        if k in challenge and challenge[k] != v:
+            remote_details_updates[k] = v
+        elif not is_default(k, v) or create_defaults:
+            remote_details_updates[k] = v
+
+    # Add all preferred keys
+    for key in preferred_order:
+        if key in remote_details_updates and key not in ignore:
+            updated_challenge[key] = remote_details_updates[key]
+
+    # Add remaining keys
+    for key in remote_details_updates:
+        if key not in preferred_order and key not in ignore:
+            updated_challenge[key] = remote_details_updates[key]
+
+    # Hack: remove tabs in multiline strings
+    updated_challenge['description'] = updated_challenge['description'].replace('\t', '')
```

Comment:
Why does this need to be done?

Reply:
Are you referring to the removal of tabs?